llmfoundry/models/hf/hf_base.py

```diff
@@ -80,19 +80,32 @@ def __init__(
         additional_train_metrics: Optional[list] = None,
         additional_eval_metrics: Optional[list] = None,
         should_save_peft_only: bool = True,
+        attn_implementation: str = 'eager',
     ):
+        if use_flash_attention_2:
+            warnings.warn(
```

Comment:
I'd rather not deprecate this actually since its a yaml change and we try very hard to not change those.